File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        // agar list khali hai toh rteun karo simple head if single elemnts thre
        if (head == nullptr || head->next == nullptr){
            return head;
        }
        // agar node khali hai or a simle elments ji same hai
        ListNode* curr = head;
        while(curr != nullptr && curr->next != nullptr){
            if(curr->val == curr->next->val){
                ListNode* duplicate = curr->next;
                curr->next = curr->next->next;
                delete duplicate; // remove memory free in c++
            }
            else{
                curr = curr->next;
            }
        }
        return head;
    }
};

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        // agar list mai single element hai yah khali hai
        if(head == null || head.next == null ){
            return head;
        }
        ListNode curr = head;
        while(curr != null && curr.next != null){
            // agar node ki value same hai agla node se
            if(curr.val==curr.next.val){
                curr.next=curr.next.next; //duplites hai
            }
            else{
                curr = curr.next;
            }
        }
        return head;
    }
}